packages/machine-extractor/src/sky/skyEnvFileUtils.ts

```diff
@@ -0,0 +1,55 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+/**
+ * Searches for the closest '.env' file starting from the directory of the given file path and moving up the directory tree until the root of the project is reached.
+ * @param filePath - The path of the file to start searching from.
+ * @param cwd - The root directory of the project.
+ * @returns The contents of the closest '.env' file.
+ * @throws An error if no '.env' file is found in the project.
+ */
+export function getClosestEnvFile({
```

Comment:
there is a handy [`find-up`](https://github.com/sindresorhus/find-up) package that you could use here, it also supports `stopAt` option that you are implementing here implicitly